src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,145 @@
+    private IntakePosition desiredPosition = IntakePosition.STOWED;
+
+    public Intake() {
+        initIntakeTalons();
+        intakePositionMap = new TreeMap<IntakePosition, Double>();
+        intakePositionMap.put(IntakePosition.DEPLOYED, IntakeCal.INTAKE_DEPLOYED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.STOWED, IntakeCal.INTAKE_STOWED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.CLEAR_OF_CONVEYOR, IntakeCal.INTAKE_SAFE_POSITION_DEGREES);
+        lastControlledTime = Optional.of(Timer.getFPGATimestamp());
+    }
+
+    public void initIntakeTalons() {
+        TalonFXConfigurator cfg = intakeTalonFront.getConfigurator();
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+        toApply.CurrentLimits.SupplyCurrentLimit = IntakeConstants.INTAKING_TALONS_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.SupplyCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        cfg.apply(toApply);
+
+        intakeTalonBack.setControl(intakeTalonBackFollower);
+    }
+
+    public double getLastControlledTime() {
+        if (lastControlledTime.isEmpty()) {
+            return Units.secondsToMilliseconds(Constants.PERIOD_TIME_SECONDS);
+        }
+        return lastControlledTime.get();
+    }
+
+    public void setLastControlledTime(double mostRecentTimestamp) {
+        lastControlledTime = Optional.of(mostRecentTimestamp);
+    }
+
+    /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+    public double getCosineArmAngle() {
+        return Math.cos(
+                Units.degreesToRadians(
+                        pivotEncoder.getPosition() - IntakeConstants.INTAKE_POSITION_WHEN_HORIZONTAL_DEGREES));
+    }
+
+    public double getOffsetAbsPositionDeg() {
+        return pivotAbsoluteEncoder.getPosition() + IntakeCal.INTAKE_ABSOLUTE_ENCODER_ZERO_OFFSET_DEG;
+      }
+
+    public void moveToPos(IntakePosition pos) {
+        pivotController.setGoal(intakePositionMap.get(pos));
+        double intakeDemandVoltsA = pivotController.calculate(getOffsetAbsPositionDeg());
+        double intakeDemandVoltsB = IntakeCal.INTAKE_PIVOT_FEEDFORWARD
+                .calculate(pivotController.getSetpoint().velocity);
+        double intakeDemandVoltsC = IntakeCal.ARBITRARY_INTAKE_PIVOT_FEEDFORWARD_VOLTS * getCosineArmAngle();
+
+        double currentTimestamp = Timer.getFPGATimestamp();
+        double timestampDifference = currentTimestamp - getLastControlledTime();
+        setLastControlledTime(currentTimestamp);
+        double intakeDemandVoltsD = IntakeCal.INTAKE_PIVOT_FEEDFORWARD.calculate(pivotController.getSetpoint().velocity,
+                pivotController.getGoal().velocity, timestampDifference);
+
+        pivotMotor.setVoltage(intakeDemandVoltsA + intakeDemandVoltsB + intakeDemandVoltsC + intakeDemandVoltsD);
+    }
+
+    public boolean atDesiredIntakePosition(IntakePosition pos) {
```

Comment:
this doesn't do desired, it checks for an input position. so the function name and the impl differ

Reply:
ill add one for both